numpy/_core/src/multiarray/datetime.c

```diff
@@ -2064,8 +2067,9 @@ NpyDatetime_ConvertPyDateTimeToDatetimeStruct(
     }
 
     /* Get the year */
-    tmp = PyObject_GetAttrString(obj, "year");
-    if (tmp == NULL) {
+    /* Checking if Exception will be raised it would be -1 else + value */
+    int res = PyObject_GetOptionalAttr(obj, npy_interned_str.year, &tmp);
+    if (res != 1) {
```

Comment:
I think this check should be combined with the `PyObject_HasAttrString` above

Reply:
thank you for the suggestion, done